pkg/ephemeral-disk/ephemeral-disk.go

```diff
@@ -130,14 +138,36 @@ func (c *ephemeralDiskCreator) CreateEphemeralImages(vmi *v1.VirtualMachineInsta
 }
 
 func createBackingDisk(backingFile string, imagePath string) ([]byte, error) {
-	// #nosec No risk for attacket injection. Parameters are predefined strings
-	cmd := exec.Command("qemu-img",
-		"create",
-		"-f",
-		"qcow2",
-		"-b",
-		backingFile,
-		imagePath,
-	)
-	return cmd.CombinedOutput()
+	if backingInfo, err := GetImageInfo(backingFile); err != nil {
+		return nil, err
+	} else {
+		// #nosec No risk for attacker injection. Parameters are predefined strings
+		cmd := exec.Command("qemu-img",
+			"create",
+			"-f",
+			"qcow2",
+			"-b",
+			backingFile,
+			"-F",
+			backingInfo.Format,
+			imagePath,
+		)
+		return cmd.CombinedOutput()
+	}
+}
+
+func GetImageInfo(imagePath string) (*DiskInfo, error) {
+	// #nosec No risk for attacker injection. Only get information about an image
+	out, err := exec.Command(
+		"/usr/bin/qemu-img", "info", imagePath, "--output", "json",
```

Comment:
Can we invoke `qemu-img` (vs. `/usr/bin/qemu-img`), as in `createBackingDisk()`, to avoid dependency on the exact filesystem path?

Reply:
Initially just copied the function "as-is". Done.